File: problems/problems_LCR_036/Solution.java
package problems.problems_LCR_036;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


public class Solution extends BaseSolution {
    public int evalRPN(String[] tokens) {
        Deque<Integer> stack = new ArrayDeque<>();
        for (String token: tokens) {
            if ("+-*/".contains(token)) {
                int num2 = stack.pop();
                int num1 = stack.pop();
                switch (token) {
                    case "+":
                        stack.push(num1 + num2);
                        break;
                    case "-":
                        stack.push(num1 - num2);
                        break;
                    case "*":
                        stack.push(num1 * num2);
                        break;
                    case "/":
                        stack.push(num1 / num2);
                        break;
                }
            } else {
                stack.push(Integer.parseInt(token));
            }
        }
        return stack.pop();
    }

    @Override
    public Object solve(String[] inputJsonValues) {
        String[] tokens = jsonArrayToStringArray(inputJsonValues[0]);
        return JSON.toJSON(evalRPN(tokens));
    }
}

File: problems/problems_LCR_036/solution.go
package problemLCR_036

import (
	"encoding/json"
	"log"
	"strconv"
	"strings"
)

func evalRPN(tokens []string) int {
	var values []int
	for _, token := range tokens {
		if strings.Contains("+-*/", token) {
			last := values[len(values)-1]
			values = values[:len(values)-1]
			switch token {
			case "+":
				values[len(values)-1] += last
			case "-":
				values[len(values)-1] -= last
			case "*":
				values[len(values)-1] *= last
			case "/":
				values[len(values)-1] /= last
			}
		} else {
			if value, err := strconv.Atoi(token); err == nil {
				values = append(values, value)
			}
		}
	}
	return values[0]
}

func Solve(inputJsonValues string) interface{} {
	inputValues := strings.Split(inputJsonValues, "\n")
	var tokens []string

	if err := json.Unmarshal([]byte(inputValues[0]), &tokens); err != nil {
		log.Fatal(err)
	}

	return evalRPN(tokens)
}
